onnxruntime/core/graph/graph_utils.cc

```diff
@@ -353,12 +353,37 @@ const ONNX_NAMESPACE::AttributeProto* GetNodeAttribute(const Node& node, const s
   return iter == attrs.end() ? nullptr : &iter->second;
 }
 
+bool IsSingleOutputUsed(const Node& node) {
```

Comment:
I think you can avoid creating a set here: get the name of the first output edge you find; if any subsequent edge does not have the same name, return false.

Reply:
You can name this IsMoreThanOneOutputUsed or IsOnlyOneOutputUsed.
IsSingleOutput implies that we have only one output.

Reply:
Maybe HasMoreThanOneOutputUsed or HasOnlyOneOutputUsed? 

---
In reply to: [283932111](https://github.com/microsoft/onnxruntime/pull/1018#discussion_r283932111) [](ancestors = 283932111)

Reply:
going with `IsOnlyOneOutputUsed` to be consistent with all the other Is* methods in graph_utils